diff --git a/linux-0.11/Makefile b/linux-0.11/Makefile
index ddb5be1..1629240 100644
--- a/linux-0.11/Makefile
+++ b/linux-0.11/Makefile
@@ -34,11 +34,11 @@ MATH	=kernel/math/math.a
 LIBS	=lib/lib.a
 
 .c.s:
-@$(CC) $(CFLAGS) -S -o $*.s $<
+	@$(CC) $(CFLAGS) -S -o $*.s $<
 .s.o:
-@$(AS)  -o $*.o $<
+	@$(AS)  -o $*.o $<
 .c.o:
-@$(CC) $(CFLAGS) -c -o $*.o $<
+	@$(CC) $(CFLAGS) -c -o $*.o $<
 
 all:	Image	
 
diff --git a/linux-0.11/include/linux/sched.h b/linux-0.11/include/linux/sched.h
index 6b6e243..9f0d3df 100644
--- a/linux-0.11/include/linux/sched.h
+++ b/linux-0.11/include/linux/sched.h
@@ -1,7 +1,7 @@
 #ifndef _SCHED_H
 #define _SCHED_H
 
-#define NR_TASKS 0
+#define NR_TASKS 5
 #define HZ 100
 
 #define FIRST_TASK task[0]
diff --git a/linux-0.11/init/main.c b/linux-0.11/init/main.c
index 8660535..c14850b 100644
--- a/linux-0.11/init/main.c
+++ b/linux-0.11/init/main.c
@@ -137,7 +137,6 @@ void main(void)		/* This really IS void, no error here. */
 	hd_init();
 	floppy_init();
 	sti();
-	panic(""); 
 	move_to_user_mode();
 	if (!fork()) {		/* we count on this going ok */
 		init();
@@ -187,9 +186,10 @@ void init(void)
 		execve("/bin/sh",argv_rc,envp_rc);
 		_exit(2);
 	}
-	if (pid>0)
+	if (pid>0) 
 		while (pid != wait(&i))
 			/* nothing */;
+	
 	while (1) {
 		if ((pid=fork())<0) {
 			printf("Fork failed in init\r\n");
diff --git a/osdi.img b/osdi.img
index a4516f6..65686c4 100644
Binary files a/osdi.img and b/osdi.img differ
diff --git a/rootfs/etc/profile b/rootfs/etc/profile
index 80df9d4..96c1995 100644
--- a/rootfs/etc/profile
+++ b/rootfs/etc/profile
@@ -10,3 +10,4 @@ alias up='cd ..'
 
 export TERM PS1 PS2
 umask 066
+echo 'Hello 0656512!!'
